models/ovm/optimism/ovm_optimism_contract_mapping.sql

```diff
@@ -0,0 +1,315 @@
+    where 
+      true
+      and success
+      and tx_success
+      and type = 'create'
+    {% if is_incremental() %} -- this filter will only be applied on an incremental run 
+      and block_time >= date_trunc('day', now() - interval '1 week')
+
+    -- to get existing history of contract mapping
+    union all 
+
+    select 
+      creator_address
+      ,contract_creator_if_factory as contract_factory
+      ,contract_address
+      ,created_time
+      ,creation_tx_hash
+      ,trace_element
+      ,is_self_destruct
+    from {{ this }}
+    {% endif %}
+  ) as x
+  group by 1, 2, 3, 4, 5, 6, 7
+)
+,tokens as (
+  select 
+    bl.contract_address
+    ,t.symbol
+  from base_level as bl 
+  join {{ ref('tokens_optimism_erc20') }} as t
+    on bl.contract_address = t.contract_address
+  group by 1, 2
+
+  union all 
+
+  select 
+    bl.contract_address
+    ,t.name as symbol
+  from base_level as bl 
+  join {{ ref('tokens_optimism_nft') }} as t
+    on bl.contract_address = t.contract_address
+  group by 1, 2
+)
+-- starting from 0 
+{% for i in range(max_levels) -%}
+,level{{i}} as (
+    select
+      {{i}} as level 
+      ,coalesce(u.creator_address, b.creator_address) as creator_address
+      {% if loop.first -%}
+      ,case
+        when u.creator_address is NULL then NULL
+        else b.creator_address
+      end as contract_factory
+      {% else -%}
+      ,case
+        when u.creator_address is NULL then b.contract_factory
+        else b.creator_address
+      end as contract_factory
+      {% endif %}
+      ,b.contract_address
+      ,b.created_time
+      ,b.creation_tx_hash
+      ,b.trace_element
+      ,b.is_self_destruct
+    {% if loop.first -%}
+    from base_level as b
+    left join base_level as u
+      on b.creator_address = u.contract_address
+    {% else -%}
+    from level{{i-1}} as b
+    left join base_level as u
+      on b.creator_address = u.contract_address
+    {% endif %}
+)
+{%- endfor %}
+
+,creator_contracts as (
+  select 
+    f.creator_address
+    ,f.contract_factory
+    ,f.contract_address
+    ,coalesce(cc.contract_project, ccf.contract_project) as contract_project 
+    ,f.created_time
+    -- check if the contract is an immediate self-destruct contract
+    ,coalesce(f.is_self_destruct, 
+      case 
+        when exists (
+          select 1 
+          from {{ source('optimism', 'traces') }} as sd 
+          where 
+            f.creation_tx_hash = sd.tx_hash
+            and f.trace_element = sd.trace_address[1]
+            and sd.type = 'suicide'
+            {% if is_incremental() %} -- this filter will only be applied on an incremental run 
+            and sd.block_time >= date_trunc('day', now() - interval '1 week')
+            {% endif %}
+        ) then true 
+      else false 
+    end) as is_self_destruct
+    ,f.creation_tx_hash
+    ,f.trace_element 
+  from level{{max_levels}} as f
+  left join {{ ref('contract_creator_address_list') }} as cc 
```

Comment:
seems like this is causing fails, should it be `ovm_optimism_contract_creator_address_list`